sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -84,43 +84,24 @@ namespace Azure { namespace Core { namespace Http {
   /**
    * HTTP request method.
    */
-  enum class HttpMethod
-  {
-    Get, ///< GET
-    Head, ///< HEAD
-    Post, ///< POST
-    Put, ///< PUT
-    Delete, ///< DELETE
-    Patch, ///< PATCH
-  };
+  class HttpMethod {
+  public:
+    HttpMethod() = delete;
+    explicit HttpMethod(std::string value) : m_value(std::move(value)) {}
+    bool operator==(const HttpMethod& other) const { return m_value == other.m_value; }
+    bool operator!=(const HttpMethod& other) const { return !(*this == other); }
+    const std::string& ToString() const { return m_value; }
+
+    AZ_CORE_DLLEXPORT const static HttpMethod Get;
```

Comment:
We are going to have a big discussion with Rick, probably on Tuesday, on using static data members vs functions in extensible enums.

Reply:
and storage team is strongly supportive of static data members. Although we're  aware of that initialization order fiasco, because we don't expect anybody to use these static variables at initialization time.